site/src/pages/release-notes/1.26.0.mdx

```diff
@@ -0,0 +1,220 @@
+  ```java
+  Server.builder()
+        .contextPath("/v1", "/v2")
+        .service(myService) // served under "/v1" and "/v2"
+        .and()
+        .virtualHost("foo.com")
+        .contextPath("/v3")
+        .service(myService) // served by virtual host "foo.com" under "/v3"
+        .build();
+  ```
+  - You can also set the context path via Spring property.
+  ```yaml
+  armeria:
+    context-path: /api/v1
+  ```
+- **Easier Request Scoping for Kotlin**: You can now easily propagate an Armeria <type://RequestContext> in Kotlin using
+  <type://ContextAwareExecutor#asCoroutineDispatcher()>. #5171
+  ```kotlin
+  withContext(ctx.eventLoop().asCoroutineDispatcher()) {
+    assert RequestContext.current() == ctx
+    ...
+  }
+  ```
+- **gRPC Exception Handler Annotation**: You can now easily handle an exception that is raised from a gRPC service using
+  <type://@GrpcExceptionHandler> annotation. #5046
+  ```java
+  @GrpcExceptionHandler(MyGrpcExceptionHandlerFunction.class)
+  @Override
+  public void unaryCall(
+    SimpleRequest request,
+    StreamObserver<SimpleResponse> responseObserver) {
+      ...
+  }
+
+  class MyGrpcExceptionHandlerFunction implements GrpcExceptionHandlerFunction {
+    @Override
+    public Status apply(RequestContext ctx, Throwable cause, Metadata metadata) {
+      if (unauthenticatedExcpetion(cause)) {
+        return Status.UNAUTHENTICATED;
+      }
+      ...
+    }
+  }
+  ```
+- **StreamMessage finalizer**: You can now produce the last value for a <type://StreamMessage> when it completes
+  exceptionally or successfully using <type://StreamMessage#endWith(Function)>. #4816 #5198
+  ```java
+  StreamMessage<Integer> message =
+    original.endWith(th -> {
+        if (th instanceof AbortedStreamException) {
+            return 100;
+        }
+        return -1;
+    });
+  ```
+- **Disable HTTP/2 for Requests**: You can now disable HTTP/2 protocol when sending HTTP or HTTPS requests. #5168
+  ```java
+  ClientFactory
+    .builder()
+    .preferHttp1(true)
+    .build();
+  ```
+- **Client Certificate Metrics**: You can now monitor client certificates with the following metrics. #5155
+  - `armeria.client.tls.certificates.validity`: `1` if valid, `0` otherwise.
+  - `armeria.client.tls.certificates.validity.days`: The number of days until the certificate expires.
+- You can now add a hook that is invoked when a <type://RequestContext> is pushed and popped. #4991 #5107
+  ```java
+  Supplier<AutoCloseable> contextHook = () -> {
+    System.out.println("Context is pushed.");
+    return () -> {
+      System.out.println("Context is popped.");
+    };
+  }
+
+  // Client
+  WebClient
+    .builder()
+    .contextHook(contextHook)
+    ...
+    .build();
+  // Server
+  Server
+    .builder()
+    .contextHook(contextHook)
+    ...
+    .build();
+  ```
+- **Notify CircuitBreak on Duration Exceed**: You can now report a failure to a circuit breaking when the total duration exceeds a certain threshold.
```

Comment:
- New Circuit Breaker Rule for Time-consuming Requests?
- We could say `New Ciercuit Breaker and Retry Rule ..` if this feature is also available for `RetryRule`